simtools/utils/general.py

```diff
@@ -913,6 +916,15 @@ def get_value_unit_type(value):
         base_value = value
         base_type = extract_type_of_value(base_value)
 
+    if unit_str is not None:
+        if base_unit is not None:
+            try:
+                base_value = base_value * u.Unit(base_unit).to(u.Unit(unit_str))
+            except u.UnitConversionError:
+                _logger.error(f"Cannot convert {base_unit} to {unit_str}.")
+                raise
+        base_unit = unit_str
+
```

Comment:
These changes made me realize that the function returns `value` and `unit`, and `value` can even a `Quantity`. This is confusing. `value` shouldn't be `Quantity`. If you need the quantity instance, then I would call it quantity instead of value.